### PR TITLE
Refactor kubeops test to use custom service names

### LINKED ISSUE #1430: [BUG] `TestKubeOpsCreateDeleteWithCoreResource` test fails intermittently on CI
KubeOpsSuite.TestKubeOpsCreateDeleteWithCoreResource should use `generateName` for the resource to avoid flakes as shown below.

```
FAIL: kubeops_test.go:248: KubeOpsSuite.TestKubeOpsCreateDeleteWithCoreResource

kubeops_test.go:263:
    c.Assert(err, IsNil, Commentf("Phase %s failed", p.Name()))
... value *errors.StatusError = &errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"Status", APIVersion:"v1"}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"services \"test-deployment-2\" already exists", Reason:"AlreadyExists", Details:(*v1.StatusDetails)(0xc000aacd20), Code:409}} ("services \"test-deployment-2\" already exists")
... Phase createDeploy failed
```

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview

This PR refactors the Kubeops testSuite

Fix #1430.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues

- #1430

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

`go test -v ./pkg/function -check.v -check.f KubeOpsSuite`
```
=== RUN   Test
PASS: kubeops_test.go:194: KubeOpsSuite.TestKubeOps     2.241s
PASS: kubeops_test.go:249: KubeOpsSuite.TestKubeOpsCreateDeleteWithCoreResource 0.119s
PASS: kubeops_test.go:286: KubeOpsSuite.TestKubeOpsCreateWaitDelete     3.332s
OK: 3 passed
--- PASS: Test (5.73s)
PASS
ok      github.com/kanisterio/kanister/pkg/function     5.754s
```

### CODE REVIEW

File: pkg/function/kubeops_test.go [hunk 1]
```diff
@@ -251,8 +252,8 @@ func (s *KubeOpsSuite) TestKubeOpsCreateDeleteWithCoreResource(c *C) {
 	tp := param.TemplateParams{}
 	action := "test"
 	gvr := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}
-	serviceName := "test-deployment-2"
-	spec := fmt.Sprintf(serviceSpec, s.namespace)
+	serviceName := rand.String(8)
```
Comment (viveksinghggits) [inline pkg/function/kubeops_test.go:255]:
```
const deploymentName = "test-deployment-2"
```
and then 
```
serviceName := fmt.Sprintf("%s-%s", deploymentName, rand.String(8))
```

File: pkg/function/kubeops_test.go [hunk 2]
```diff
@@ -205,10 +208,10 @@ func (s *KubeOpsSuite) TestKubeOps(c *C) {
 		expResource resourceRef
 	}{
 		{
-			bp: newCreateResourceBlueprint(createInSpecsNsPhase(s.namespace)),
+			bp: newCreateResourceBlueprint(createInSpecsNsPhase(testServiceName, s.namespace)),
```
Comment (pavannd1) [inline pkg/function/kubeops_test.go:211]: We should add random string to this name also

File: pkg/function/kubeops_test.go [hunk 3]
```diff
@@ -195,54 +198,71 @@ func (s *KubeOpsSuite) TestKubeOps(c *C) {
...
 	for _, tc := range []struct {
+		name        string
 		bp          crv1alpha1.Blueprint
 		expResource resourceRef
+		errChecker  Checker
 	}{
 		{
-			bp: newCreateResourceBlueprint(createInSpecsNsPhase(s.namespace)),
+			// Do not expect the resource to be created with the provided name
+			name: fmt.Sprintf("%s-%s", testServiceName, rand.String(8)),
```
Comment (viveksinghggits) [inline pkg/function/kubeops_test.go:214]: which name is this? is this the same name that we are creating the resource with?

**Review by ihcsim — APPROVED** [review]
LGTM - thanks :+1:.

**Pull request merged**